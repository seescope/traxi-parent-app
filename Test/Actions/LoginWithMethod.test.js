import loginWithMethod from '../../App/Actions/LoginWithMethod';
import { NativeModules } from 'react-native';

describe('LoginWithMethod', () => {
  it('logs the user in, saves the profile in AsyncStorage, then dispatches LOGGED_IN', (done) => {
    // Setup
    const TEST_METHOD = 'anonymous';
    const TEST_PROFILE = { name: 'TEST' };

    const mockDispatch = jest.fn();
    const mockAuthenticate = jest.fn(() => Promise.resolve(TEST_PROFILE));

    NativeModules.Authentication = {
      authenticate: mockAuthenticate,
    };

    // Exercise
    return loginWithMethod(TEST_METHOD)(mockDispatch).then(profile => {
      expect(mockAuthenticate).toHaveBeenCalled();
      expect(mockDispatch.mock.calls).toMatchSnapshot();
      expect(profile).toEqual(TEST_PROFILE);

      done();
    }).catch(e => {
      done.fail(e);
    });
  });
});

