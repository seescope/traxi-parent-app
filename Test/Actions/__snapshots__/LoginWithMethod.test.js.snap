// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWithMethod logs the user in, saves the profile in AsyncStorage, then dispatches LOGGED_IN 1`] = `
Array [
  Array [
    Object {
      "profile": Object {
        "name": "TEST",
      },
      "type": "LOGGED_IN",
    },
  ],
]
`;
